### PR TITLE
Click helper

### PR DESCRIPTION
**Pull request by nemchik**
In a previous attempt to submit way too much, one of the comments had mentioned a click helper function that could aid in simplifying the code. This pull satisfies that idea.
The clickHelper function is able to fully replace the simulateClick function, but both are still included and will work side by side.
I have included SoundCloud as an example of the clickHelper where there is no frame involved in the player.
I have included Spotify as an example of the clickHelper where there is a frame housing the player.
Both players have been tested.

### CODE REVIEW

File: extension/shared.js [hunk 1]
```diff
@@ -33,6 +33,26 @@ function simulateClick(element) {
     return element.dispatchEvent(click);
 }
 
+function clickHelper(element, frame) {
```
Comment (borismus) [inline extension/shared.js]: Would prefer a more descriptive name, like simulateSelectorClick

**Comment by borismus** [comment]
Great, thanks. Could you migrate to the helper throughout?

**Comment by nemchik** [comment]
I've migrated to the helper in all the sockets except those that called an
element by an index (see keysocket-bop.js use of [0]. I'm not sure how this
would be handled by the helper and I've never used that player myself).


--
Eric Nemchik
http://nemchik.com/

**Comment by nemchik** [comment]
Something broke (worked yesterday). I'm going to delete this pull and make a new one once I've got it working.

**Pull request closed without merging**